receiver/mongodbreceiver/config.go

```diff
@@ -59,3 +61,40 @@ func (c *Config) Validate() error {
 
 	return err
 }
+
+func (c *Config) ClientOptions() *options.ClientOptions {
+	clientOptions := options.Client()
+	connString := fmt.Sprintf("mongodb://%s", strings.Join(c.hostlist(), ","))
+	clientOptions.ApplyURI(connString)
+
+	if c.Timeout > 0 {
+		clientOptions.SetConnectTimeout(c.Timeout)
+	}
+
+	tlsConfig, err := c.LoadTLSConfig()
+	if err != nil && tlsConfig != nil {
+		clientOptions.SetTLSConfig(tlsConfig)
+	}
+
+	if c.ReplicaSet != "" {
+		clientOptions.SetReplicaSet(c.ReplicaSet)
+	}
+
+	if c.Username != "" && c.Password != "" {
+		authOpt := options.Credential{
+			Username: c.Username,
+			Password: c.Password,
+		}
+		clientOptions.SetAuth(authOpt)
```

Comment:
nit: Typically don't bother saving the var if you use it immediately and only once. The exception is when the line would be too verbose/long, but this the following is typical.
```suggestion
		clientOptions.SetAuth(options.Credential{
			Username: c.Username,
			Password: c.Password,
		})
```